### PR TITLE
feat: 617 - new "save packagings V3" feature

### PR DESCRIPTION
**Pull request by monsieurtanuki**
New files:
* `api_saveProductV3_test.dart`: Integration tests around the "save packagings V3" feature.
* `ProductPackaging.dart`: Packaging component for a product, e.g. recyclable bottle made of glass.
* `ProductPackaging.g.dart`: generated

Impacted files:
* `AbstractQueryConfiguration.dart`: populated parameter `tags_lc`
* `api_getProduct_test.dart`: removed references to now deprecated field `packaging`; added a test about the new packagings field
* `api_saveProduct_test.dart`: skipped one 504 test; removed references to deprecated field `packaging`
* `HttpHelper.dart`: new method `doPatchRequest`; minor refactoring
* `openfoodfacts.dart`: new method `temporarySaveProductV3`
* `Product.dart`: deprecated field `packaging`; added field `packagings`
* `Product.g.dart`: generated
* `ProductFields.dart`: deprecated `PACKAGING`; added `PACKAGINGS` and `RAW`
* `ProductSearchQueryConfiguration.dart`: added api version parameter
* `UriHelper.dart`: new method `getPatchUri`

### What
- Here we implement the new "save packagings V3" feature.
- The new `packagings` product field provides detailed information about the packaging, cf. new class `ProductPackaging`
- Unfortunately for the moment only `packagings` is supported with save/V3. That means that packagings should be saved with a different call from the other fields.

### Fixes bug(s)
- Closes: #617

### CODE REVIEW

File: lib/model/Product.dart [hunk 1]
```diff
@@ -311,8 +312,14 @@ class Product extends JsonObject {
       includeIfNull: false)
   Map<OpenFoodFactsLanguage, List<String>>? labelsTagsInLanguages;
 
+  // TODO: deprecated from 2022-12-16; remove when old enough
+  @Deprecated('User packagingS instead')
```
Comment (M123-dev) [inline lib/model/Product.dart:316]:
```suggestion
  // TODO: deprecated from 2022-12-16; remove when old enough
  @Deprecated('Use [packagings] instead')
```

File: lib/model/Product.dart [hunk 2]
```diff
@@ -468,7 +475,8 @@ class Product extends JsonObject {
       this.labels,
       this.labelsTags,
       this.labelsTagsInLanguages,
-      this.packaging,
+      // TODO: deprecated from 2022-12-16; remove when old enough
+      @Deprecated('Use packagingS field instead') this.packaging,
```
Comment (M123-dev) [inline lib/model/Product.dart:479]:
```suggestion
      @Deprecated('Use [packagings] field instead') this.packaging,
```

Feel free to leave it with the capital S

File: lib/openfoodfacts.dart [hunk 3]
```diff
@@ -147,6 +148,48 @@ class OpenFoodAPIClient {
     return Status.fromApiResponse(response.body);
   }
 
+  /// Temporary: saves product packagings V3 style.
+  ///
+  /// For the moment that's the only field supported in WRITE by API V3.
+  /// Long term target is of course more something like [saveProduct].
+  static Future<ProductResultV3> temporarySaveProductV3(
```
Comment (M123-dev) [inline lib/openfoodfacts.dart:155]:
```suggestion
  static Future<ProductResultV3> temporarySaveProductPackagingsV3( 
```
Just a suggestion

File: lib/utils/HttpHelper.dart [hunk 4]
```diff
@@ -29,10 +29,10 @@ class HttpHelper {
   static const String FROM = 'anonymous';
 
   /// Adds user agent data to parameters, for statistics purpose
-  static Map<String, String>? addUserAgentParameters(
-    Map<String, String>? map,
+  static Map<String, dynamic>? addUserAgentParameters(
+    Map<String, dynamic>? map,
   ) {
-    map ??= <String, String>{};
+    map ??= <String, dynamic>{};
```
Comment (M123-dev) [inline lib/utils/HttpHelper.dart:35]: Why dynamic

**Review by M123-dev — APPROVED** [review]
Looks good, thanks @monsieurtanuki just some minor comments

File: lib/openfoodfacts.dart (hunk 3 above)
Comment (monsieurtanuki) [reply lib/openfoodfacts.dart:155]: Could make sense but in the long run I expect more fields to match V3 (e.g. one by month) : I'm not going to create a different method for each.

File: lib/model/Product.dart (hunk 2 above)
Comment (monsieurtanuki) [reply lib/model/Product.dart:479]: The square bracket syntax is unfortunately not detected in strings, only in `///` comments.

File: lib/utils/HttpHelper.dart (hunk 4 above)
Comment (monsieurtanuki) [reply lib/utils/HttpHelper.dart:35]: With the new method `doPatchRequest` the input can be `Map<String, dynamic>`, and not `Map<String, String>`.
More specifically in that case:
```dart
    parameterMap['product']['packagings'] = packagings;
```

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev for your fast review!

**Pull request merged**